### DISCUSSION TITLE
Leaflet VS Mapbox VS Mapbox GL VS ...

### DISCUSSION THREAD

**Discussion by francoiscrouzet** [archaiodata/thacer#81] [General]
@AlZorba 
Slightly related to https://github.com/archaiodata/thacer/issues/58

We currently use leaflet 0.7.7
current version is 1.9.3
Docs for all versions: https://leafletjs.com/reference-versions.html

mapbox pricing: https://www.mapbox.com/pricing

Pas mal de retour d'utilisateurs ici :
https://stackoverflow.com/questions/35069753/mapbox-gl-js-vs-mapbox-js
https://stackoverflow.com/questions/48812974/mapboxgl-js-versus-leaflet-js

Quelques "les meilleurs lib de map gnagnagna"
https://colorlib.com/wp/javascript-libraries-for-creating-dynamic-maps/
https://dev.to/camptocamp-geo/the-3-best-open-source-web-mapping-libraries-57o7
https://www.google.com/search?q=open+source+free+map+javascript+library&client=firefox-b-d&source=lnt&tbs=qdr:y&sa=X&ved=2ahUKEwiLxbqu9-r9AhWkTaQEHeMTA4gQpwV6BAgBEB4&biw=1352&bih=758&dpr=2

Les 2 githubs, avec les issues, les fork, les stars
https://github.com/mapbox/mapbox-gl-js
https://github.com/Leaflet/Leaflet

**Comment by AlZorba** [comment]
super job.
Je suis conforté dans Leaflet mais Mapbox resterait gratuit à notre niveau et Openlayer est la solution utilisée par l'EfA et Dipylon je crois.
Tu as testé les versions plus récente de Leaflet? je regarde ça par curiosité.

**Reply by francoiscrouzet** [reply]
Hello, oui les liens que je t'ai envoyé pousse effectivement vers leaflet. Et ce sans compter le code et les competences deja existantes.
Openlayer, peut-être peux-tu demander à l'occasion pourquoi ce choix?
Non pas regardé versions plus recente, j'ai vu qu'il a changé la casse de certaines fonctions, et que la fonction `collapse` qui pourrait nous servir pour le responsif n'est pas dans notre version

**Reply by francoiscrouzet** [reply]
ah merci pour le "super job"! :)

**Comment by AlZorba** [comment]
Looki,g first at removing Mapbox dependencies for the tile layers.
Many options listed : https://leaflet-extras.github.io/leaflet-providers/preview/

**Reply by francoiscrouzet** [reply]
Hum ok, i was not aware of this problematic, you'll show me during a call maybe

**Comment by chtzv** [comment]
@AlZorba , @francoiscrouzet ,
Do you have any further thoughts on moving away from Mapbox? It seems to me that everything that has been done so far with Mapbox is achievable with Leaflet only. But reworking the code won't be that simple. It will be a major overhaul of the create-layer functions. It can't be done gradually because - as you noted - Mapbox loads a very old version of Leaflet, which lacks functionality we need. I tried to load the most recent Leaflet parallel to Mapbox but it does not work. 
So I did a bit of experiment and removed Mapbox completely, loading the latest Leaflet only. I rewrote the functions depending on Mapbox, and I made 5 out of the 7 layers work. I am having difficulties with 'Secteurs de fouille et GTh' and 'Echantillons géologiques'. Would you like me to continue working on this? Or do you consider different options for replacing Mapbox?

**Reply by AlZorba** [reply]
@chtzv 
That is terrific if you can upgrade Leaflet and get ride of Mapbox!
I understand there is a big gap but if after all what you have done only two layers are broken, maybe I can work on them.
What are the trouble on those layers?

**Comment by chtzv** [comment]
Good! I will open an issue and a branch so we can work together on this.

**Reply by francoiscrouzet** [reply]
Hey ! This is very linked with that : https://github.com/archaiodata/thacer/issues/84
Not sure if it should be done at the same time, but it should be considered I think
I am a bit sick today, but will be available later today for translation

**Comment by chtzv** [comment]
@AlZorba , @francoiscrouzet , check branch #103 now. Mapbox is gone and all layers work. There are some changes in the appearance but  these can be adjusted (discuss first?). As a positive side effect, some of the errors that populated the console previously are gone.

Problems:
If you click on a sector the function searchCeramByClick will throw an error 'featureLayerCeram is undefined'. I can investigate further, but I wonder whether it would be better to address this as part of #84.

**Reply by francoiscrouzet** [reply]
Hey @chtzv, seems great ! 
I am not sure how to check diffs easily on a branch. 
Is there any easy possiblity? If not, would you mind creating a PR already, but making it a draft (not mergeable yet) if necessary ?

**Reply by chtzv** [reply]
@francoiscrouzet , I fixed the searchCeramByClick error. The function was using a Mapbox method. I also opened a draft PR.

**Reply by AlZorba** [reply]
@chtzv seems huge step forward. I won't be able to test before Tuesday, so I'll let you know.

**Reply by francoiscrouzet** [reply]
Same thing  here, not available today, will have a look tomorrow

**Comment by AlZorba** [comment]
Choi
